src/main/blackbox/blackbox_io.c

```diff
@@ -145,12 +145,24 @@ void blackboxWriteUnsignedVB(uint32_t value)
 }
 
 /**
+ * ZigZag encoding maps all values of a signed integer into those of an unsigned integer in such
+ * a way that numbers of small absolute value correspond to small integers in the result.
+ *
+ * (Compared to just casting a signed to an unsigned which creates huge resulting numbers for
+ * small negative integers).
+ */
+static uint32_t zigzagEncode(int32_t value)
+{
+    return (uint32_t)((value << 1) ^ (value >> 31));
+}
+
```

Comment:
Yeah being able to decode the data back to the original is a requirement too, but we also need to ensure that the bytes on the wire are what we document them to be, so that if someone writes their own decoder following the spec, they'll decode the same result. If we only test our encoder/decoder together then they could be writing something completely different to what we thought we were writing.